src/uu/ls/src/ls.rs

```diff
@@ -929,19 +929,19 @@ impl Config {
 
         let width = match options.get_one::<String>(options::WIDTH) {
             Some(x) => parse_width(x)?,
-            None => match terminal_size::terminal_size() {
-                Some((width, _)) => width.0,
-                None => match std::env::var_os("COLUMNS") {
-                    Some(columns) => match columns.to_str().and_then(|s| s.parse().ok()) {
-                        Some(columns) => columns,
-                        None => {
-                            show_error!(
-                                "ignoring invalid width in environment variable COLUMNS: {}",
-                                columns.quote()
-                            );
-                            DEFAULT_TERM_WIDTH
-                        }
-                    },
+            None => match std::env::var_os("COLUMNS") {
```

Comment:
maybe move this block into a new function?

Reply:
I just finished refactoring this

Reply:
new PR is fine :)

Reply:
okay i will do that